MmSupervisorPkg/Test/MmSupvRequestUnitTestApp/Acpi.c

```diff
@@ -0,0 +1,192 @@
+/** @file
+
+Modified from Tianocore
+https://github.com/tianocore/edk2-platforms/tree/master/Silicon/Intel/IntelSiliconPkg/Feature/VTd/IntelVTdDxe
+for purpose of easy ACPI table parsing
+
+  Copyright (c) 2017 - 2018, Intel Corporation. All rights reserved.<BR>
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PciSegmentLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiLib.h>
+
+#include <Guid/Acpi.h>
+
+#include <Protocol/PciIo.h>
+#include <Protocol/PciEnumerationComplete.h>
+#include <Protocol/IoMmu.h>
+
+#include <IndustryStandard/Pci.h>
+#include <IndustryStandard/DmaRemappingReportingTable.h>
+
+#include "Acpi.h"
+
+/**
+  This function scan ACPI table in RSDT.
+
+  @param[in]  Rsdt      ACPI RSDT
+  @param[in]  Signature ACPI table signature
+
+  @return ACPI table
+**/
+VOID *
+ScanTableInRSDT (
+  IN RSDT_TABLE  *Rsdt,
+  IN UINT32      Signature
+  )
+{
+  UINTN                        Index;
+  UINT32                       EntryCount;
+  UINT32                       *EntryPtr;
+  EFI_ACPI_DESCRIPTION_HEADER  *Table;
+
+  EntryCount = (Rsdt->Header.Length - sizeof (EFI_ACPI_DESCRIPTION_HEADER)) / sizeof (UINT32);
+
+  EntryPtr = &Rsdt->Entry;
+  for (Index = 0; Index < EntryCount; Index++, EntryPtr++) {
+    Table = (EFI_ACPI_DESCRIPTION_HEADER *)((UINTN)(*EntryPtr));
+    if ((Table != NULL) && (Table->Signature == Signature)) {
+      return Table;
+    }
+  }
+
+  return NULL;
+}
+
+/**
+  This function scan ACPI table in XSDT.

```

Comment:
This function returns a pointer to the ACPI table entry that matches the passed Signature

Reply:
Thanks. This is updated.